lib/hanami/assets/compilers/sass.rb

```diff
@@ -27,37 +27,63 @@ def self.eligible?(name)
         # @api private
         def initialize(*)
           super
-          require "sassc"
+          require "sass-embedded"
+          require "uri"
         end
 
         private
 
         # @since 0.3.0
         # @api private
         def renderer
-          @renderer ||=
-            ::SassC::Engine.new(
-              source.read,
-              syntax: target_syntax,
-              load_paths: load_paths
-            )
+          @renderer ||= SassRenderer.new(source, load_paths: load_paths)
         end
 
         # @since 0.3.0
         # @api private
         def dependencies
-          renderer.dependencies.map(&:filename)
-        rescue source::NotRenderedError
-          []
+          renderer.dependencies
         end
 
-        # @since 1.3.2
+        # @since 2.0.0
         # @api private
-        def target_syntax
-          if source.extname =~ /sass\z/.freeze
-            :sass
-          else
-            :scss
+        class SassRenderer
+          # @since 2.0.0
+          # @api private
+          def initialize(source, **kwargs)
+            @source = source
+            @kwargs = kwargs
+          end
+
+          # @since 2.0.0
+          # @api private
+          def render
+            @result = ::Sass.compile(@source, **@kwargs)
+            @result.css
+          end
+
+          # @since 2.0.0
+          # @api private
+          def dependencies
+            dependencies = []
+            return dependencies unless @result
+
+            @result.loaded_urls.each do |url|
+              if url.start_with?("file:")
+                dependencies << URI.parse(url).route_from(root_url).to_s
+              end
+            end
+
+            dependencies
+          end
+
+          # @since 2.0.0
+          # @api private
+          def root_url
+            @root_url ||= begin
+              root = Hanami.respond_to?(:root) ? Hanami.root : Dir.pwd
```

Comment:
Not sure if this code copied from similar code elsewhere is the correct way to get project root.